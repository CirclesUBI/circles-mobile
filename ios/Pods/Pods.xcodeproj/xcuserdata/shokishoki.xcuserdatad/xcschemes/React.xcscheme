<?xml version="1.0" encoding="UTF-8"?>
<Scheme
   LastUpgradeVersion = "0930"
   version = "1.3">
   <BuildAction
      parallelizeBuildables = "YES"
      buildImplicitDependencies = "YES">
      <BuildActionEntries>
         <BuildActionEntry
            buildForAnalyzing = "YES"
            buildForTesting = "YES"
            buildForRunning = "YES"
            buildForProfiling = "YES"
            buildForArchiving = "YES">
            <BuildableReference
               BuildableIdentifier = "primary"
               BlueprintIdentifier = "E1FE208787E57E5DCF17D5EA6C6B8065"
               BlueprintName = "React"
               ReferencedContainer = "container:Pods.xcodeproj"
               BuildableName = "libReact.a">
            </BuildableReference>
         </BuildActionEntry>
      </BuildActionEntries>
   </BuildAction>
   <TestAction
      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
      shouldUseLaunchSchemeArgsEnv = "YES"
      buildConfiguration = "Debug">
      <AdditionalOptions>
      </AdditionalOptions>
   </TestAction>
   <LaunchAction
      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
      launchStyle = "0"
      useCustomWorkingDirectory = "NO"
      ignoresPersistentStateOnLaunch = "NO"
      debugDocumentVersioning = "YES"
      debugServiceExtension = "internal"
      buildConfiguration = "Debug"
      allowLocationSimulation = "YES">
      <AdditionalOptions>
      </AdditionalOptions>
   </LaunchAction>
   <ProfileAction
      savedToolIdentifier = ""
      useCustomWorkingDirectory = "NO"
      debugDocumentVersioning = "YES"
      buildConfiguration = "Release"
      shouldUseLaunchSchemeArgsEnv = "YES">
   </ProfileAction>
   <AnalyzeAction
      buildConfiguration = "Debug">
   </AnalyzeAction>
   <ArchiveAction
      buildConfiguration = "Release"
      revealArchiveInOrganizer = "YES">
   </ArchiveAction>
</Scheme>
