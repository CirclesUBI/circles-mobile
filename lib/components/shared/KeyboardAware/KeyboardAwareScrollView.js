import React from 'react'
import PropTypes from 'prop-types'

import { ScrollView } from 'react-native'

import KeyboardAwareBase from './KeyboardAwareBase'
import { defaultTheme } from 'uPortMobile/lib/styles'

export default class KeyboardAwareScrollView extends KeyboardAwareBase {
  render() {
    const { styles } = this.context.theme ? this.context.theme : defaultTheme
    return (
      <ScrollView 
        {...this.props}
        contentContainerStyle={styles.contentContainer}
        style={styles.container}
        contentInset={{bottom: this.state.keyboardHeight}}
        ref={(r) => {
          this._keyboardAwareView = r
        }}
        onLayout={(layoutEvent) => {
          this._onKeyboardAwareViewLayout(layoutEvent.nativeEvent.layout)
        }}
        onScroll={(event) => {
          this._onKeyboardAwareViewScroll(event.nativeEvent.contentOffset)
          if (this.props.onScroll) {
            this.props.onScroll(event)
          }
        }}
        onContentSizeChange={() => {
          this._updateKeyboardAwareViewContentSize()
        }}
        scrollEventThrottle={200}
      />
    )
  }
}

KeyboardAwareScrollView.propTypes = {
  getTextInputRefs: PropTypes.func,
  onScroll: PropTypes.func
}

KeyboardAwareScrollView.defaultProps = {
  ...KeyboardAwareBase.defaultProps,
  getTextInputRefs: () => {
    return []
  }
};

KeyboardAwareScrollView.contextTypes = {
  theme: PropTypes.object
}
