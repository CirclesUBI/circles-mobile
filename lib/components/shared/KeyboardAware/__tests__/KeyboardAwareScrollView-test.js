import { ScrollView } from 'react-native'
import React from 'react'
import KeyboardAwareScrollView from '../KeyboardAwareScrollView'
import renderer from 'react-test-renderer'
// 21,22,26,42
describe('KeyboardAwareScrollView', () => {
  it('renders the KeyboardAwareScrollView', () => {
    const tree = renderer.create(
      <KeyboardAwareScrollView
        dataSource={{rowIdentities: []}}
        />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onKeyboardAwareViewLayout from onLayout', () => {
    this.wrapper = global.shallow(
      <KeyboardAwareScrollView
        dataSource={{rowIdentities: []}}
        />
    )
    const instance = this.wrapper.instance()
    instance._onKeyboardAwareViewLayout = jest.fn()
    this.wrapper.find(ScrollView).props().onLayout({nativeEvent: true})
    expect(instance._onKeyboardAwareViewLayout).toHaveBeenCalled()
  })
  it('calls onKeyboardAwareViewScroll from onScroll', () => {
    const onScroll = jest.fn()
    this.wrapper = global.shallow(
      <KeyboardAwareScrollView
        dataSource={{rowIdentities: []}}
        onScroll={onScroll}
        />
    )
    const instance = this.wrapper.instance()
    instance._onKeyboardAwareViewScroll = jest.fn()
    this.wrapper.find(ScrollView).props().onScroll({nativeEvent: true})
    expect(instance._onKeyboardAwareViewScroll).toHaveBeenCalled()
    expect(onScroll).toHaveBeenCalled()
  })
  it('calls updateKeyboardAwareViewContentSize from onContentSizeChange', () => {
    this.wrapper = global.shallow(
      <KeyboardAwareScrollView
        dataSource={{rowIdentities: []}}
        />
    )
    const instance = this.wrapper.instance()
    instance._updateKeyboardAwareViewContentSize = jest.fn()
    this.wrapper.find(ScrollView).props().onContentSizeChange()
    expect(instance._updateKeyboardAwareViewContentSize).toHaveBeenCalled()
  })
  it('returns an empty array from getTextInputRefs', () => {
    this.wrapper = global.shallow(
      <KeyboardAwareScrollView
        dataSource={{rowIdentities: []}}
        />
    )
    expect(this.wrapper.props().getTextInputRefs()).toEqual([])
  })
})
