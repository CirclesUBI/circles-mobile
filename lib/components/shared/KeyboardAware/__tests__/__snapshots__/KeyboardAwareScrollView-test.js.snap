// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardAwareScrollView renders the KeyboardAwareScrollView 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "padding": 15,
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  dataSource={
    Object {
      "rowIdentities": Array [],
    }
  }
  getTextInputRefs={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onScroll={[Function]}
  scrollEventThrottle={200}
  scrollToBottomOnKBShow={false}
  scrollToInputAdditionalOffset={75}
  startScrolledToBottom={false}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;
